include/extractor/guidance/suppress_mode_handler.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef OSRM_EXTRACTOR_GUIDANCE_SUPPRESS_MODE_HANDLER_HPP_
+#define OSRM_EXTRACTOR_GUIDANCE_SUPPRESS_MODE_HANDLER_HPP_
+
+#include "extractor/guidance/constants.hpp"
+#include "extractor/guidance/intersection.hpp"
+#include "extractor/guidance/intersection_generator.hpp"
+#include "extractor/guidance/intersection_handler.hpp"
+#include "extractor/travel_mode.hpp"
+#include "util/node_based_graph.hpp"
+
+namespace osrm
+{
+namespace extractor
+{
+namespace guidance
+{
+
+class SuppressModeHandler : public IntersectionHandler
```

Comment:
I think you can make it `SuppressModeHandler final : ` - all handler could be final except the base class `IntersectionHandler`